src/dmd/cppmangle.d

```diff
@@ -1315,7 +1315,18 @@ private final class CppMangleVisitor : Visitor
 
         foreach (n, fparam; parameterList)
         {
-            Type t = target.cpp.parameterType(fparam);
+            Type t = fparam.type.merge2();
+            if (fparam.isReference())
+                t = t.referenceTo();
+            else if (fparam.storageClass & STC.lazy_)
+            {
+                // Mangle as delegate
+                auto tf = new TypeFunction(ParameterList(), t, LINK.d);
+                auto td = new TypeDelegate(tf);
+                t = td.merge();
+            }
+            else if (Type cpptype = target.cpp.parameterType(t))
+                t = cpptype;
```

Comment:
This added line does not seem to be needed and it is not covered.

Reply:
It is covered by gdc, and you will also see it hit if you have `extern(C++) int ice(int[1])` - but this is not suitable for the testsuite because it results in an ICE.